BtConnectorLib/btconnectorlib2/src/main/java/org/thaliproject/p2p/btconnectorlib/BluetoothBase.java

```diff
@@ -35,53 +31,58 @@ public BluetoothBase(Context Context, BluetoothStatusChanged handler) {
 
         //bluetooth = new BluetoothAdapter(this);
         bluetooth = BluetoothAdapter.getDefaultAdapter();
-        if(bluetooth != null) {
+        if (bluetooth != null) {
             blueAddress = bluetooth.getAddress();
         }
     }
 
-    public boolean Start() {
+    public synchronized  boolean Start() {
+
+        if (bluetooth == null) {
+            return false;
+        }
 
-        boolean ret = false;
-        if(bluetooth != null) {
-            ret = true;
-            Log.d("", "Start-My BT: " + blueAddress);
+        Log.d("", "Start-My BT: " + blueAddress);
 
-            if(receiver == null) {
-                filter = new IntentFilter();
-                filter.addAction(BluetoothAdapter.ACTION_SCAN_MODE_CHANGED);
+        if (receiver == null) {
+            try {
                 receiver = new BtBrowdCastReceiver();
+                IntentFilter filter = new IntentFilter();
+                filter.addAction(BluetoothAdapter.ACTION_SCAN_MODE_CHANGED);
                 this.context.registerReceiver(receiver, filter);
-            }
+            } catch(Exception e) {e.printStackTrace();}
```

Comment:
I changed that to return false to indicate that we did not start bluetooth, thus bluetooth is considered not Ok.